src/instructlab/taxonomy/diff.py

```diff
@@ -81,16 +81,20 @@ def diff(ctx, taxonomy_path, taxonomy_base, yaml_rules, quiet):
 
     # validate new or changed taxonomy files
     try:
-        validate_taxonomy(taxonomy_path, taxonomy_base, yaml_rules)
+        valid = validate_taxonomy(taxonomy_path, taxonomy_base, yaml_rules)
     except (TaxonomyReadingException, yaml.YAMLError) as exc:
         if not quiet:
             click.secho(
                 f"Reading taxonomy failed with the following error: {exc}",
                 fg="red",
             )
+        else:
+            logger.debug(f"Reading taxonomy failed with the following error: {exc}")
         raise SystemExit(1) from exc
     if not quiet:
-        click.secho(
-            f"Taxonomy in {taxonomy_path} is valid :)",
-            fg="green",
-        )
+        if valid:
+            click.secho(f"Taxonomy in {taxonomy_path} is valid :)", fg="green")
+        else:
+            click.secho(
+                f"Taxonomy in {taxonomy_path} contains no qna.yaml files.", fg="yellow"
```

Comment:
I think a challenge here is providing a proper, actionable message. It is not that there are no qna.yaml files (there may be many), but that in comparing the taxonomy folder with the [base branch](https://github.com/instructlab/instructlab/blob/69e22831070624997120f94b68b48569a111313e/src/instructlab/configuration.py#L255-L258) (default is [origin/main](https://github.com/instructlab/instructlab/blob/69e22831070624997120f94b68b48569a111313e/src/instructlab/defaults.py#L64)) there we no differences in qna.yaml files (either in uncommitted files or commits made to the current branch being something other than the taxonomy base branch.)

Reply:
Thanks. 
Is it ok for `"No new or changed qna.yaml files compared to the base branch {taxonomy_base}`?